src/main/java/org/folio/services/instance/InstanceService.java

```diff
@@ -64,7 +64,16 @@ public Future<Response> createInstance(Instance entity) {
           InstanceStorage.PostInstanceStorageInstancesResponse.class, postResponse);
 
         return postResponse.future()
-          .compose(domainEventPublisher.publishCreated());
+          .compose(response -> {
+            // Return the response without waiting for a domain event publish
+            // to complete. Units of work performed by this service is the same
+            // but the ordering of the units of work provides a benefit to the
+            // api client invoking this endpoint. The response is returned
+            // a little earlier so the api client can continue its processing
+            // while the domain event publish is satisfied.
+            domainEventPublisher.publishCreated().apply(response);
+            return Future.succeededFuture(response);
+          });
```

Comment:
A more idiomatic way is using

    .onSuccess(response -> {

so that no `return` is needed.

Reply:
Thank you. I will keep this in mind next time!